### ISSUE TITLE
USB ethernet adapter repeatedly drops in 5.19.0-76051900-generic

### ISSUE THREAD

**Issue by ozdreamern** [pop-os/linux#172]
OS: Pop!_OS 22.04 LTS x86_64 
Host: Lemur Pro lemp9 
Kernel: 5.19.0-76051900-generic 

Since the update to 5.19.0, I've experienced repeated disconnects of my USB ethernet port, usually a few minutes after starting to use it -- not specifically after it's first initialized, but rather a few minutes after putting it under some load (refreshing a heavy website, running Zoom).  When this happens, the wifi connection (AX201) remains up and stable.

The device is an ASIX AX88179 USB 3.0 Gigabit Ethernet adapter, which I've used reliably for years on PopOS without prior issue.

Below are log entries from the last cold boot through the time of the drops (some address numbers redacted).  Initially I ran a ping to a server for ~35 minutes without doing anything else, and the wired connection stayed up.  After that, I started up Firefox and  surfed the web, and the wired ethernet link went down within 3-4 minutes.

Note: I  didn't find anything more useful by setting Network Manager log domains ether & core to level "debug", but I am probably not investigating this the right way.  Let me know if I can offer additional info.

```
ug 23 12:40:44 lemp9 kernel: [    3.089594] ax88179_178a 2-1.3.1:1.0 enx0050ffff41b1: renamed from eth0
Aug 23 12:40:44 lemp9 NetworkManager[823]: <info>  [1661272844.4930] manager: (enx0050ffff41b1): new Ethernet device (/org/freedesktop/NetworkManager/Devices/2)
Aug 23 12:40:44 lemp9 NetworkManager[823]: <info>  [1661272844.4933] device (enx0050ffff41b1): state change: unmanaged -> unavailable (reason 'managed', sys-iface-state: 'external')
Aug 23 12:40:45 lemp9 networkd-dispatcher[845]: ERROR:Unknown state for interface NetworkctlListState(idx=2, name='enx0050ffff41b1', type='ether', operational='n/a', administrative='unmanaged'): n/a
Aug 23 12:40:48 lemp9 kernel: [   17.535119] ax88179_178a 2-1.3.1:1.0 enx0050ffff41b1: ax88179 - Link status is: 1
Aug 23 12:40:48 lemp9 kernel: [   17.542463] IPv6: ADDRCONF(NETDEV_CHANGE): enx0050ffff41b1: link becomes ready
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2060] device (enx0050ffff41b1): carrier: link connected
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2087] device (enx0050ffff41b1): state change: unavailable -> disconnected (reason 'carrier-changed', sys-iface-state: 'managed')
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2116] device (enx0050ffff41b1): Activation: starting connection 'Wired connection 1' (6e6eff91-9fcf-3b23-87bd-4e1ed69ec540)
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2120] device (enx0050ffff41b1): state change: disconnected -> prepare (reason 'none', sys-iface-state: 'managed')
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2127] device (enx0050ffff41b1): state change: prepare -> config (reason 'none', sys-iface-state: 'managed')
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2141] device (enx0050ffff41b1): state change: config -> ip-config (reason 'none', sys-iface-state: 'managed')
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2152] dhcp4 (enx0050ffff41b1): activation: beginning transaction (timeout in 45 seconds)
Aug 23 12:40:48 lemp9 avahi-daemon[828]: Joining mDNS multicast group on interface enx0050ffff41b1.IPv6 with address fe80::x.y.z:e13f.
Aug 23 12:40:48 lemp9 avahi-daemon[828]: New relevant interface enx0050ffff41b1.IPv6 for mDNS.
Aug 23 12:40:48 lemp9 avahi-daemon[828]: Registering new address record for fe80::x.y.z:e13f on enx0050ffff41b1.*.
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2221] dhcp4 (enx0050ffff41b1): state changed new lease, address=172.x.y.107
Aug 23 12:40:48 lemp9 avahi-daemon[828]: Joining mDNS multicast group on interface enx0050ffff41b1.IPv4 with address 172.x.y.107.
Aug 23 12:40:48 lemp9 avahi-daemon[828]: New relevant interface enx0050ffff41b1.IPv4 for mDNS.
Aug 23 12:40:48 lemp9 avahi-daemon[828]: Registering new address record for 172.x.y.107 on enx0050ffff41b1.IPv4.
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2240] device (enx0050ffff41b1): state change: ip-config -> ip-check (reason 'none', sys-iface-state: 'managed')
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2265] device (enx0050ffff41b1): state change: ip-check -> secondaries (reason 'none', sys-iface-state: 'managed')
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2268] device (enx0050ffff41b1): state change: secondaries -> activated (reason 'none', sys-iface-state: 'managed')
Aug 23 12:40:48 lemp9 systemd-resolved[737]: enx0050ffff41b1: Bus client set default route setting: yes
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2285] device (enx0050ffff41b1): Activation: successful, device activated.
Aug 23 12:40:48 lemp9 systemd-resolved[737]: enx0050ffff41b1: Bus client set DNS server list to: 1.1.1.1, 1.0.0.1
Aug 23 12:40:48 lemp9 NetworkManager[823]: <info>  [1661272848.2673] policy: set 'Wired connection 1' (enx0050ffff41b1) as default for IPv4 routing and DNS
Aug 23 12:45:48 lemp9 NetworkManager[823]: <info>  [1661273148.2412] dhcp4 (enx0050ffff41b1): state changed new lease, address=172.x.y.107
Aug 23 12:50:48 lemp9 NetworkManager[823]: <info>  [1661273448.2297] dhcp4 (enx0050ffff41b1): state changed new lease, address=172.x.y.107
Aug 23 12:55:48 lemp9 NetworkManager[823]: <info>  [1661273748.2313] dhcp4 (enx0050ffff41b1): state changed new lease, address=172.x.y.107
Aug 23 13:00:48 lemp9 NetworkManager[823]: <info>  [1661274048.2285] dhcp4 (enx0050ffff41b1): state changed new lease, address=172.x.y.107
Aug 23 13:05:48 lemp9 NetworkManager[823]: <info>  [1661274348.2255] dhcp4 (enx0050ffff41b1): state changed new lease, address=172.x.y.107
Aug 23 13:10:48 lemp9 NetworkManager[823]: <info>  [1661274648.2259] dhcp4 (enx0050ffff41b1): state changed new lease, address=172.x.y.107
Aug 23 13:15:48 lemp9 NetworkManager[823]: <info>  [1661274948.2287] dhcp4 (enx0050ffff41b1): state changed new lease, address=172.x.y.107
Aug 23 13:25:48 lemp9 NetworkManager[823]: <info>  [1661275548.2175] dhcp4 (enx0050ffff41b1): activation: beginning transaction (timeout in 45 seconds)
Aug 23 13:25:48 lemp9 NetworkManager[823]: <info>  [1661275548.2176] dhcp4 (enx0050ffff41b1): state changed no lease
Aug 23 13:25:49 lemp9 avahi-daemon[828]: Withdrawing address record for 172.x.y.107 on enx0050ffff41b1.
Aug 23 13:25:49 lemp9 avahi-daemon[828]: Leaving mDNS multicast group on interface enx0050ffff41b1.IPv4 with address 172.x.y.107.
Aug 23 13:25:49 lemp9 avahi-daemon[828]: Interface enx0050ffff41b1.IPv4 no longer relevant for mDNS.
Aug 23 13:26:33 lemp9 NetworkManager[823]: <info>  [1661275593.7184] device (enx0050ffff41b1): state change: activated -> failed (reason 'ip-config-unavailable', sys-iface-state: 'managed')
Aug 23 13:26:33 lemp9 NetworkManager[823]: <warn>  [1661275593.7200] device (enx0050ffff41b1): Activation: failed for connection 'Wired connection 1'
Aug 23 13:26:33 lemp9 NetworkManager[823]: <info>  [1661275593.7225] device (enx0050ffff41b1): state change: failed -> disconnected (reason 'none', sys-iface-state: 'managed')
Aug 23 13:26:33 lemp9 avahi-daemon[828]: Withdrawing address record for fe80::x.y.z:e13f on enx0050ffff41b1.
Aug 23 13:26:33 lemp9 avahi-daemon[828]: Leaving mDNS multicast group on interface enx0050ffff41b1.IPv6 with address fe80::x.y.z:e13f.
Aug 23 13:26:33 lemp9 avahi-daemon[828]: Interface enx0050ffff41b1.IPv6 no longer relevant for mDNS.
Aug 23 13:26:33 lemp9 NetworkManager[823]: <info>  [1661275593.7583] dhcp4 (enx0050ffff41b1): canceled DHCP transaction
Aug 23 13:26:33 lemp9 systemd-resolved[737]: enx0050ffff41b1: Bus client set default route setting: no
Aug 23 13:26:33 lemp9 systemd-resolved[737]: enx0050ffff41b1: Bus client reset DNS server list.
Aug 23 13:26:33 lemp9 NetworkManager[823]: <info>  [1661275593.7883] device (enx0050ffff41b1): Activation: starting connection 'Wired connection 1' (6e6eff91-9fcf-3b23-87bd-4e1ed69ec540)
Aug 23 13:26:33 lemp9 NetworkManager[823]: <info>  [1661275593.7884] device (enx0050ffff41b1): state change: disconnected -> prepare (reason 'none', sys-iface-state: 'managed')
Aug 23 13:26:33 lemp9 NetworkManager[823]: <info>  [1661275593.7886] device (enx0050ffff41b1): state change: prepare -> config (reason 'none', sys-iface-state: 'managed')
Aug 23 13:26:33 lemp9 NetworkManager[823]: <info>  [1661275593.7889] device (enx0050ffff41b1): state change: config -> ip-config (reason 'none', sys-iface-state: 'managed')
Aug 23 13:26:33 lemp9 NetworkManager[823]: <info>  [1661275593.7893] dhcp4 (enx0050ffff41b1): activation: beginning transaction (timeout in 45 seconds)
Aug 23 13:26:33 lemp9 avahi-daemon[828]: Joining mDNS multicast group on interface enx0050ffff41b1.IPv6 with address fe80::x.y.z:e13f.
```

Thanks for reading!

**Comment by ozdreamern** [comment]
After searching through lkml and patchwork messages and commits, I found more relevant info on this problem:

https://bugzilla.kernel.org/show_bug.cgi?id=216327
https://lkml.org/lkml/2022/8/19/1382
https://bugs.archlinux.org/task/75491

It looks like the [problematic patch](https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net.git/commit/?id=36a15e1cb134) got backed out in 5.19.2, so I'm expecting (hoping) it will be stable again in the next Pop_OS! production kernel (looks like 5.19.5 is currently being tested).

**Comment by ozdreamern** [comment]
Closing this as fixed in 5.19.16.
